Fix panic when waiting for pod times out

MikaelSmith opened the pull request:
Always check for an error before trying to cast the result, as casting a nil causes a panic. Also ensure the pod is deleted if there's an error waiting for it to be available. `kubectl get events` may help in identifying why the pod didn't start.

Fixes #774.

Signed-off-by: Michael Smith <michael.smith@puppet.com>

MikaelSmith committed:
Fix panic when waiting for pod times out

Always check for an error before trying to cast the result, as casting a
nil causes a panic. Also ensure the pod is deleted if there's an error
waiting for it to be available. `kubectl get events` may help in
identifying why the pod didn't start.

Fixes #774.

Signed-off-by: Michael Smith <michael.smith@puppet.com>

ekinanp commented on plugin/kubernetes/pvc.go:
```diff
@@ -135,13 +135,15 @@ func (v *pvc) inContainer(ctx context.Context, fn containerCb) (interface{}, err
 		}
+		cleanup = func() {
+			// Use a background context to ensure deletion happens even if context was cancelled.
+			activity.Record(ctx, "Deleted temporary pod %v: %v", tempPod, tempPod.delete(context.Background()))
```
Why background? Why not a timeout context?

MikaelSmith replied:
Hmm. I guess. I'd hope the Delete implementation would have a somewhat bounded runtime, especially if it also supports context cancellation.

MikaelSmith replied:
Same comment would also apply to https://github.com/puppetlabs/wash/blob/master/plugin/docker/volume.go#L122-L129.

ekinanp replied:
Ah yeah, I missed that one. I guess the Q. here is "Does this code introduce a theoretically-possible edge-case of a dangling go-routine?" If so, may as well fix it since adding a timeout context is easy. If not, then no need.

Either way not a blocker. Rest of the PR looks good to me.

MikaelSmith replied:
The timeout context is no guarantee either, it'll only halt the goroutine if the code that's blocking also checks the context for cancelation.

MikaelSmith committed:
Avoid identifying data race in test setup

Tests would periodically fail with
```
WARNING: DATA RACE
Write at 0x00c000190790 by goroutine 80:
  github.com/puppetlabs/wash/transport.(*SSHTestSuite).SetupTest()
      /home/runner/work/wash/wash/transport/ssh_test.go:128 +0x52
  github.com/stretchr/testify/suite.Run.func2()
      /home/runner/go/pkg/mod/github.com/stretchr/testify@v1.5.1/suite/suite.go:123 +0x630
  testing.tRunner()
      /opt/hostedtoolcache/go/1.14.2/x64/src/testing/testing.go:991 +0x1eb

Previous read at 0x00c000190790 by goroutine 77:
  github.com/stretchr/testify/mock.(*Mock).findExpectedCall()
      /home/runner/go/pkg/mod/github.com/stretchr/testify@v1.5.1/mock/mock.go:267 +0x41
  github.com/stretchr/testify/mock.(*Mock).MethodCalled()
      /home/runner/go/pkg/mod/github.com/stretchr/testify@v1.5.1/mock/mock.go:348 +0xb3
  github.com/stretchr/testify/mock.(*Mock).Called()
      /home/runner/go/pkg/mod/github.com/stretchr/testify@v1.5.1/mock/mock.go:338 +0x1e4
  github.com/puppetlabs/wash/transport.(*SSHTestSuite).Handler()
      /home/runner/work/wash/wash/transport/ssh_test.go:44 +0xc3
  github.com/puppetlabs/wash/transport.(*SSHTestSuite).Handler-fm()
      /home/runner/work/wash/wash/transport/ssh_test.go:43 +0x26
  github.com/gliderlabs/ssh.(*session).handleRequests.func1()
      /home/runner/go/pkg/mod/github.com/gliderlabs/ssh@v0.3.0/session.go:239 +0x5c
```

Signed-off-by: Michael Smith <michael.smith@puppet.com>
